rohmu/delta/common.py

```diff
@@ -143,11 +143,18 @@ class SnapshotFile(DeltaModel):
     mtime_ns: int
     hexdigest: str = ""
     content_b64: Optional[str]
+    # Indicator if a file should be part of a chunk/bundle - useful for the files which are small to be treated as
+    # separate delta hash files, but not small enough to be embedded into a manifest file, so better to group them
+    # together when e.g. uploading to the object storage
+    is_bundled: bool = False
 
     def __lt__(self, o):
         # In our use case, paths uniquely identify files we care about
         return self.relative_path < o.relative_path
 
+    def __hash__(self):
+        return hash(self.relative_path)
```

Comment:
SnapshotFile instances are mutable. Allowing hashing of a mutable instance is always a peril. Unless there's a very, very good reason, which should come with specific "here be dragons" doc, I'd not do that. At least ```relative_path``` must be immutable if you want to use it as an hash key

Reply:
As you can see by the function before, we identify the uniqueness by the relative_path and in practice we never change that, let me think how I can do that immutable.

Reply:
This is the implementation today. But the interface does not prevent changes. Anybody making a (possibly legit) change in the future could introduce a subtle bug.

I think pydantic supports immutable models, but I don't know if you can set just some fields to be immutable. If it where a normal class, I'd just say "use a _property_name and expose a readonly property".

Reply:
I did that already, please take a look